src/server.cpp

```diff
@@ -40,23 +41,31 @@ void Server::asyncSend(const Packet& response, const std::function<void(const er
        [this, callback](const error_code& ec, std::size_t /*bytesTransferred*/) {handleSend(ec, callback);});
 }
 
-void Server::handleReceive(const error_code& error, std::size_t bytes, std::function<void(const error_code&, const Packet&)> callback)
+void Server::handleReceive(const error_code& error, std::size_t bytes, std::function<void(const error_code&, const std::optional<Packet>&)> callback)
 {
-    try
-    {
-        if (error)
-            throw std::runtime_error{"Error async_receivw_from: " + error.message()};
+        std::optional<Packet> oEmpty;
+        std::optional<Packet> oRequest;
 
         if (bytes < 20)
-            throw std::runtime_error{"Error: request length less than 20 bytes"};
+            callback(lib::Error::numberBytesLess20, oEmpty);
 
+    try
+    {
         const Packet request = Packet(m_recvBuffer, bytes, m_secret);
-        callback(error, request);
+        oRequest = request;
     }
     catch (const std::runtime_error& exception)
     {
-        return;
+        if (std::string (exception.what()) == "lib::Error::numberBytesLess20")
```

Comment:
Never ever use string dispatch. Instead, as we discussed above, catch your dedicated exception class and extract `error_code` from it directly.